index.html

```diff
@@ -4292,6 +4306,144 @@ <h3>Proof Formats</h3>
       </section>
     </section>
 
+    <section class="normative">
+      <h2>Algorithms</h2>
+
+      <p>
+This section contains algorithms that can be used by implementations to
+perform common operations.
+      </p>
+
+      <section class="normative">
+        <h3>Verification</h3>
+
+        <p>
+This section contains an algorithm that <a>conforming verifier
+implementations</a> can run when verifying a <a>verifiable credential</a> or a
+<a>verifiable presentation</a>. This algorithm takes a byte sequence (bytes
+<var>inputBytes</var>) and a media type (string <var>inputMediaType</var>) as
+inputs, and produces an object (<var>verified</var>) that contains the
+following:
+        </p>
+
+        <ul>
+          <li>
+a verification status (boolean <var>verified</var>.<var>status</var>)
+          </li>
+          <li>
+a verified document (object <var>verified</var>.<var>document</var>)
+          </li>
+          <li>
+a media type (string <var>verified</var>.<var>mediaType</var>)
+          </li>
+          <li>
+zero or more optional warnings (array of <a>anomalyDescriptions</a> <var>verified</var>.<var>warnings</var>)
+          </li>
+          <li>
+zero or more optional errors (array of <a>anomalyDescriptions</a> <var>verified</var>.<var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>verified</var> to an empty object.
+          </li>
+          <li>
+Verify that the securing mechanism in use on <var>inputBytes</var> has not been
+tampered with, using <var>inputMediaType</var> as additional input to the
+algorithm. See Section <a href="#securing-mechanisms"></a>
+for details.
+          </li>
+          <li>
+If the verification of the securing mechanism is successful, set the following:
+            <ul>
+              <li>
+<var>verified</var>.<var>status</var> to `true`
+              </li>
+              <li>
+<var>verified</var>.<var>document</var> to the information that was checked and
+returned as successfully protected by the verification process for the securing
+mechanism
+              </li>
+              <li>
+<var>verified</var>.<var>mediaType</var> to the corresponding
+media type
+              </li>
+              <li>
+<var>verified</var>.<var>warnings</var> optionally to any warnings returned by
+the verification process for the securing mechanism
+              </li>
+            </ul>
+            <ol class="algorithm">
+              <li>
+Check whether <var>verified</var>.<var>document</var> is a
+<a>conforming document</a>. If not, optionally add appropriate warnings
+and errors to <var>verified</var>.<var>warnings</var> and
+<var>verified</var>.<var>errors</var>, respectively.
```

Comment:
I think the proposed algorithm has a mismatch with the current definition of `verification`. 

Currently, the [definition](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verify) says

> The evaluation of whether a [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) or [verifiable presentation](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-presentation) is an authentic and timely statement of the issuer or presenter, respectively. This includes checking that: the credential (or presentation) conforms to the specification; the proof method is satisfied; and, if present, the status check succeeds. Verification of a credential does not imply evaluation of the truth of [claims](https://www.w3.org/TR/vc-data-model-2.0/#dfn-claims) encoded in the credential..

With the proposed algorithm, the `verified` object return may look as follows:
```json
{
  "status":"true",
  "document": {},
  "mediaType":"",
  "errors": [
    "The document is not conforming. It's missing credentialSubject"
  ]
}
```

Such object has a "verified" status, but the document is not conformant. This is directly at odds with the definition of `verification`.

I think `status` should reflect checking conformance AND also status checks. Alternatively, we can update the definitions of `verification`.

Reply:
I suggest we just remove status checking from the verification step. That seems more like a validation thing (I know it's debatable either way). /cc @jandrieu

Reply:
I think that makes sense for status checking. That still leaves conformance. Can a non-conforming document result in `verified.status == true`? I think the answer should be no.

Reply:
Correct, the answer there is "no" -- a non-conforming document will not result in `verified.status === true` ... I've done a series of updates to this PR to make that more clear.